datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/ci.py

```diff
@@ -75,7 +81,11 @@ def validate_master_jobs(fix, repo_data, testable_checks, cached_display_names):
     fixed = False
 
     for job in jobs:
-        check_name = job['checkName']
+        check_name = job.get('checkName')
+        if not check_name:
+            job = list(job.values())[0][0]
+            check_name = job.get('checkName')
```

Comment:
WDYTA a small dedicated util to explain the intent behind this `list(...)[0][0]` usage? We do it twice in this file.

```python
def get_check_name_from_job(job):
    try:
        return job['checkName']
    except KeyError:
        # Probably a nested block using an AZP template variable:
        # - ${{ if ... }}:
        #    - checkName: ...
        job = list(job.values())[0][0]
        return job['checkName']
```

Then this would become:

```suggestion
        check_name = get_check_name_from_job(job)
```